spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultJdbcClient.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.lock;
+
+import java.util.Date;
+import java.util.UUID;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+@Component
+@Transactional
+class DefaultJdbcClient implements JdbcClient {
```

Comment:
OK. I'm fine with the solution. Although I still think that it can't be `@Component` unconditionally.
Consider the `@ComponentScan` and `@Bean` definition. Then `@Autowired` by type in the target `JdbcLockRegistry`.

OTOH this class has to be `public`. I like the extracted interface which opens a window for any customizations in the target application. Although I don't like the name. 
Maybe `JdbcLockRegistryHelper`, `JdbcLockService`, `JdbcLockRepository`? But not just `JdbcClient`. to broad in context of just `lock` logic.